test/integration/camera.cc

```diff
@@ -568,32 +649,110 @@ void CameraSensorTest::CameraProjection(const std::string &_renderEngine)
   while (!done && sleep++ < maxSleep)
   {
     std::lock_guard<std::mutex> lock(g_mutex);
-    done = (camera1Counter > 0 && camera2Counter > 0);
+    done = (camera1Counter > 0u && camera2Counter > 0u && camera3Counter > 0u &&
+        camera1DataCounter > 0u && camera2DataCounter > 0u &&
+        camera3DataCounter > 0u);
     std::this_thread::sleep_for(std::chrono::milliseconds(100));
   }
 
   // Image size, focal length and optical center
   // Camera sensor without projection tag
-  double error = 1e-1;
-  EXPECT_EQ(camera1Info.width(), 1000u);
-  EXPECT_EQ(camera1Info.height(), 1000u);
-  EXPECT_NEAR(camera1Info.projection().p(0), 863.22975158691406, error);
-  EXPECT_NEAR(camera1Info.projection().p(5), 863.22975158691406, error);
+  // account for error converting gl projection values back to
+  // cv projection values
+  double error = 1;
+  EXPECT_EQ(camera1Info.width(), width);
+  EXPECT_EQ(camera1Info.height(), width);
+  EXPECT_NEAR(camera1Info.projection().p(0), 866.23, error);
+  EXPECT_NEAR(camera1Info.projection().p(5), 866.23, error);
   EXPECT_DOUBLE_EQ(camera1Info.projection().p(2), 500);
   EXPECT_DOUBLE_EQ(camera1Info.projection().p(6), 500);
   EXPECT_DOUBLE_EQ(camera1Info.projection().p(3), 0);
   EXPECT_DOUBLE_EQ(camera1Info.projection().p(7), 0);
 
   // Camera sensor with projection tag
-  EXPECT_EQ(camera2Info.width(), 1000u);
-  EXPECT_EQ(camera2Info.height(), 1000u);
+  EXPECT_EQ(camera2Info.width(), height);
+  EXPECT_EQ(camera2Info.height(), height);
   EXPECT_DOUBLE_EQ(camera2Info.projection().p(0), 866.23);
-  EXPECT_DOUBLE_EQ(camera2Info.projection().p(5), 966.23);
+  EXPECT_DOUBLE_EQ(camera2Info.projection().p(5), 866.23);
   EXPECT_DOUBLE_EQ(camera2Info.projection().p(2), 500);
-  EXPECT_DOUBLE_EQ(camera2Info.projection().p(6), 400);
+  EXPECT_DOUBLE_EQ(camera2Info.projection().p(6), 500);
```

Comment:
```suggestion
  EXPECT_DOUBLE_EQ(camera2Info.projection().p(6), 500.0);
```

Reply:
maybe we should add `.0` to the other values

Reply:
added. [0d7a5ff](https://github.com/gazebosim/gz-sensors/pull/314/commits/0d7a5ff4e92cbe643cf81d2d0db03bd6f56ba771)